### PR TITLE
Add a basic test suite for `TokenAuthority`

### PR DESCRIPTION
**Pull request by tarek-y-ismail**
Just the tests I could come up with as I wrote. Feel free to provide edge cases or other test cases.

### CODE REVIEW

File: tests/unit-tests/shell/test_token_authority.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    ASSERT_FALSE(revoked1);
+    ASSERT_FALSE(revoked2);
+
+    token_authority.revoke_token(token1);
+
+    auto empty_token_opt = token_authority.get_token_for_string(std::string(token1));
+    ASSERT_FALSE(empty_token_opt.has_value());
+    ASSERT_TRUE(revoked1);
+    ASSERT_FALSE(revoked2);
```
Comment (mattkae) [inline tests/unit-tests/shell/test_token_authority.cpp:140]: Nit: Should we assert that the `token2` is still present as well?

**Review by mattkae — APPROVED** [review]
One small nit pick, but I sure do like tests :)
